tilelang/cuda/language/print.py

```
@@ -0,0 +1,207 @@
+        buffer (tirx.Buffer): The buffer whose values need to be printed.
+        elems (int): The number of elements in the buffer to print.
+    """
+    if condition:
+        # Iterate through the buffer elements and print each one.
+        for i in serial(elems):
+            coords = index_to_coordinates(i, buffer.shape)
+            tirx.call_extern("handle", "debug_print_buffer_value", msg, buffer.name, i, buffer[coords])
+
+
+# NOTE(chaofan): T.print is implemented as a macro, so no return
+def print(obj: Any = None, msg: str = "", warp_group_id: int = 0, warp_id: int = 0) -> None:
```

Ruff (0.15.21)

[error] Variable `print` is shadowing a Python builtin

(A001)